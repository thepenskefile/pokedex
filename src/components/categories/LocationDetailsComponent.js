// @flow

import React from 'react';
import { Container, Flex, Heading, Set, Text, Tag } from 'fannypack';

type Props = {
  data: Object
};

const LocationDetailsComponent = ({ data }: Props) => (
  <Container textAlign="left" marginLeft="10px" width="96%">
    <Flex alignItems="flex-start" marginBottom="10px">
      <Heading>{data.name.charAt(0).toUpperCase() + data.name.slice(1)}</Heading>
    </Flex>
    {data.region && (
      <Set marginBottom="20px" spacing="major-2" display="block">
        <Text use="strong">Region: </Text>
        {data.region.name}
      </Set>
    )}

    <Set marginBottom="20px">
      {data.game_indices.map((index, i) => (
        <Tag
          key={index.generation.url.replace(/\D/g, '').substring(1)}
          backgroundColor={`generation-${index.generation.url.replace(/\D/g, '').substring(1)}`}
          size="medium"
        >
          {`gen ${index.generation.url.replace(/\D/g, '').substring(1)}`}
        </Tag>
      ))}
    </Set>
  </Container>
);

export default LocationDetailsComponent;
