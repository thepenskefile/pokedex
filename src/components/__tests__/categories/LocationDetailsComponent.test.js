import React from 'react';
import LocationDetailsComponent from '../../categories/LocationDetailsComponent';
import renderer from 'react-test-renderer';

const defaultProps = {
  data: {
    name: 'mockName',
    region: 'mockRegion'
  }
};
it('renders correctly', () => {
  const component = renderer.create(<LocationDetailsComponent {...defaultProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
