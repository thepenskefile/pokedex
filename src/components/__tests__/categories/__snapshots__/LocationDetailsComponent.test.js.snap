// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="styled-sc-144wh3b-0 iUlbjp Box-sc-6sssmg-0 inpIDs"
  style={
    Object {
      "marginLeft": "10px",
      "textAlign": "left",
      "width": "96%",
    }
  }
  width="96%"
>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
  >
    Name: 
    mockName
  </div>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
  >
    Region: 
  </div>
</div>
`;
